src/NLog/Logger.cs

```diff
@@ -82,6 +82,11 @@ protected internal Logger()
         /// </summary>
         public LogFactory Factory { get; private set; }
 
+        /// <summary>
+        /// Properties added with <see cref="WithProperty"/> or <see cref="SetProperty"/>
+        /// </summary>
+        public IDictionary<string, object> Properties => _contextProperties ?? new Dictionary<string, object>();
```

Comment:
You are opening a door to race-condition-hell by returning an unprotected dictionary.

Reply:
And that race condition isn't there with WithProperty/SetProperty

Reply:
I know ` WithProperty/SetProperty` are nice and sweet (because I made them so). But the evil user that starts calling the IDictionary-inteface to insert/remove elements will break everything.

Reply:
Sorry, I don't see what it should break. Could you please give an example ?

Reply:
Very confused that you don't know what happens if one enumerates a collection while another modifies the collection. Made some pseudo-code in new comment.